changelog.d/11742.misc

```diff
@@ -0,0 +1 @@
+Minor efficiency improvements when inserting many values into the database.
```

Comment:
So I guess the efficiency savings here are:
- the interpreter doesn't have to have to have `PyObject*` pointers repeatedly pointing to the same set of keys (column names)
  - not sure if it'd repeatedly allocate for copies of the same string literals?
- the interpreter doesn't have to spend time calculating hashes, resizing the hash table, ... and any other fun involved in maintaining a dictionary
- we don't have to spend time doing sorting in the code removed in https://github.com/matrix-org/synapse/pull/11742/commits/36919b1f415cdc895548c253d4583fe8380201ca, and we save a function call too

Reply:
I was mostly thinking of the allocation of all the dictionaries, but yes to all of that!

I also copied this changelog from the previous PR.